core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/exceptions/HashProviderException.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.user.core.exceptions;
+
+/**
+ * This class contains the implementation of HashProviderException.
+ */
+public class HashProviderException extends Exception {
+
+    private static final long serialVersionUID = -6057036683816666355L;
+    private String errorCode;
```

Comment:
I think we need to initialize the `errorCode`. Otherwise it will lead for errors if we call `getErrorCode`. WDYT?

Reply:
Let's fix in other places as well

Reply:
Initialized.